### PR TITLE
Update/onboarding medium importer style

### PR DESCRIPTION
**Pull request by zaerl**
#### Changes proposed in this Pull Request

This PR inherits logic behind wp-admin `ImportingPane` and `UploadingPane` to add additional styles we find in TAK5FpwTvfbENJDfHc7Aq1-fi.

* Add additional style changes to Medium import workflow
* New `ImportingPane` component
* New `UploadingPane` component
* WPCOM `WPCOM_REST_API_V2_Endpoint_Is_Site_Importable::is_site_importable` will returns additional data. Add a `favicon` and `title` string fields to `UrlData` type.

#### Testing instructions

* Go to: http://calypso.localhost:3000/start/importer/capture?siteSlug=YOUR_SITE
* Input a Medium file URL such as https://medium.com/@worthy_shadows_goose_107
* Load a medium `.zip` file
* See that the visual changes are the same as the TAK5FpwTvfbENJDfHc7Aq1-fi

#### Important
The author list component is missing the favicon because Medium `analyze-call` returns `null`. Try for example `https://medium.com/@worthy_shadows_goose_107`.

This can be added in a new PR.

Related to #59036.
Related to D72416-code.

### CODE REVIEW

**Comment by zaerl** [comment]
@jeroenpf [UrlData TS type](https://github.com/Automattic/wp-calypso/blob/d52afdde467625cae45bd3b61583b789f1976f3e/client/signup/steps/import/types.ts#L10-L13) is changed in order to reflect the backend D72416-code changes. See https://github.com/Automattic/wp-calypso/issues/59630

File: client/signup/steps/import-from/components/importing-pane/importing-pane.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		if ( this.isFinished() ) {
+			percentComplete = 100;
+			statusMessage = this.getSuccessText();
+		}
+
+		if ( this.isImporting() && hasProgressInfo( progress ) ) {
+			const remainingResources = resourcesRemaining( progress );
+			percentComplete = calculateProgress( progress );
+			blockingMessage = this.getImportMessage( remainingResources );
+		}
```
Comment (bogiii) [inline client/signup/steps/import-from/components/importing-pane/importing-pane.tsx:51]: Just a suggestion, maybe, this selected code block could be divided into multiple methods so the code could be more readable. Such as `getProgress`, `getStatusMessage`, `getBlockingMessage`.

What do you think?

**Comment by bogiii** [comment]
There are still missing styles to match with the Figma design.

Screenshot 2022-01-12 at 11 06 05

Screenshot 2022-01-12 at 11 18 55
There is an adjusted style of the Progress component for the onboarding flow, you can check it here https://github.com/Automattic/wp-calypso/blob/update/onboarding-medium-importer-style/client/signup/steps/import-from/wix/index.tsx#L128-L138.

Screenshot 2022-01-12 at 11 22 14

My thought is that all the mentioned stuff can be fixed by applying proper CSS.

**Review by bogiii — APPROVED** [review]
I've tested it, it looks good!
Styles are matched with the design, well done.

🚢

**Pull request merged**